### ISSUE TITLE
Adiar fails to compile

### ISSUE THREAD

**Issue by lassemoldrup** [SSoelvsten/adiar#403]
**Describe the bug**
Simply running `make` on my Mac (macOS version 12.4) fails to compile giving the following error:
```
[ 51%] Linking CXX static library libtpie.a
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libtpie.a(portability.cpp.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: libtpie.a(portability.cpp.o) has no symbols
[ 51%] Built target tpie
[ 53%] Building CXX object src/adiar/CMakeFiles/adiar.dir/adiar.cpp.o
In file included from /Users/lasse/Code/adiar/src/adiar/adiar.cpp:1:
In file included from /Users/lasse/Code/adiar/src/adiar/adiar.h:24:
In file included from /Users/lasse/Code/adiar/src/adiar/../adiar/zdd.h:476:
/Users/lasse/Code/adiar/src/adiar/../adiar/zdd/build.h:113:22: error: no matching function for call to 'min'
      id_t curr_id = std::min(remaining_levels, max_id);
                     ^~~~~~~~
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:2532:1: note: candidate template ignored: deduced conflicting types for parameter '_Tp'
      ('unsigned long' vs. 'unsigned long long')
min(const _Tp& __a, const _Tp& __b)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:2543:1: note: candidate template ignored: could not match 'initializer_list<type-parameter-0-0>' against
      'unsigned long'
min(initializer_list<_Tp> __t, _Compare __comp)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:2523:1: note: candidate function template not viable: requires 3 arguments, but 2 were provided
min(const _Tp& __a, const _Tp& __b, _Compare __comp)
^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/algorithm:2552:1: note: candidate function template not viable: requires single argument '__t', but 2 arguments
      were provided
min(initializer_list<_Tp> __t)
^
1 error generated.
make[5]: *** [src/adiar/CMakeFiles/adiar.dir/adiar.cpp.o] Error 1
make[4]: *** [src/adiar/CMakeFiles/adiar.dir/all] Error 2
make[3]: *** [src/adiar/CMakeFiles/adiar.dir/rule] Error 2
make[2]: *** [adiar] Error 2
make[1]: *** [build/static] Error 2
make: *** [build] Error 2
```
`cmake --version`: `cmake version 3.24.1`
`clang --version`: `Apple clang version 11.0.3 (clang-1103.0.32.59)`

**To Reproduce**
N/A

**Additional context**
The project has previously compiled, and I don't think I have done anything except update the library.

**Comment by SSoelvsten** [comment]
@lassemoldrup , can you check whether branch `bug/317` addresses this issue?

**Comment by lassemoldrup** [comment]
@SSoelvsten That seems to have fixed the error, but there is now a new one:
```
[ 56%] Building CXX object src/adiar/CMakeFiles/adiar.dir/bdd/apply.cpp.o
In file included from /Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:9:
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/levelized_priority_queue.h:43:20: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
  constexpr size_t no_lookahead_bound(const size_t degree = 1)
                   ^
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/levelized_priority_queue.h:45:16: note: non-constexpr function 'pow<int, unsigned long>' cannot be used in a constant
      expression
    return 8 * std::pow(2, degree);
               ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/math.h:995:1: note: declared here
pow(_A1 __lcpp_x, _A2 __lcpp_y) _NOEXCEPT
^
In file included from /Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:10:
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:296:28: warning: suggest braces around initialization of subobject [-Wmissing-braces]
          prod_pq_1.push({ r.t1, r.t2, NIL });
                           ^~~~~~~~~~
                           {         }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:595:14: note: in instantiation of function template specialization
      'adiar::__product_construction<adiar::apply_prod_policy, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 0, internal_sorter,
      internal_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>, adiar::internal_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
      return __product_construction<prod_policy,
             ^
/Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:111:12: note: in instantiation of function template specialization 'adiar::product_construction<adiar::apply_prod_policy>'
      requested here
    return product_construction<apply_prod_policy>(bdd_1, bdd_2, op);
           ^
In file included from /Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:10:
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:359:28: warning: suggest braces around initialization of subobject [-Wmissing-braces]
          prod_pq_2.push({ t1, t2, v0.low, v0.high, source });
                           ^~~~~~
                           {     }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:359:28: warning: suggest braces around initialization of subobject [-Wmissing-braces]
          prod_pq_2.push({ t1, t2, v0.low, v0.high, source });
                           ^~~~~~~~~~~~~~~~~~~~~~~
                           {                      }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:102:24: warning: suggest braces around initialization of subobject [-Wmissing-braces]
      prod_pq_1.push({ target.t1, target.t2, source });
                       ^~~~~~~~~~~~~~~~~~~~
                       {                   }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:288:9: note: in instantiation of function template specialization
      'adiar::prod_recurse_out<adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 0, internal_sorter, internal_priority_queue,
      adiar::node_file, 2, std::__1::less<unsigned int>, 1> >' requested here
        prod_recurse_out(prod_pq_1, aw, op, out_uid, r.low);
        ^
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:595:14: note: in instantiation of function template specialization
      'adiar::__product_construction<adiar::apply_prod_policy, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 0, internal_sorter,
      internal_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>, adiar::internal_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
      return __product_construction<prod_policy,
             ^
/Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:111:12: note: in instantiation of function template specialization 'adiar::product_construction<adiar::apply_prod_policy>'
      requested here
    return product_construction<apply_prod_policy>(bdd_1, bdd_2, op);
           ^
In file included from /Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:10:
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:149:24: warning: suggest braces around initialization of subobject [-Wmissing-braces]
      prod_pq_1.push({ r.t1, r.t2, source });
                       ^~~~~~~~~~
                       {         }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:112:19: note: in instantiation of function template specialization
      'adiar::prod_recurse_in__forward::go<adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 0, internal_sorter,
      internal_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1> >' requested here
      out_policy::go(prod_pq_1, aw, ea, prod_pq_1.pull().source);
                  ^
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:397:11: note: in instantiation of function template specialization
      'adiar::prod_recurse_in<adiar::prod_recurse_in__forward, adiar::prod_rec_skipto, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label,
      adiar::tuple_fst_lt, 0, internal_sorter, internal_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>,
      adiar::internal_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
          prod_recurse_in<prod_recurse_in__forward>(prod_pq_1, prod_pq_2, aw, r, t1, t2);
          ^
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:595:14: note: in instantiation of function template specialization
      'adiar::__product_construction<adiar::apply_prod_policy, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 0, internal_sorter,
      internal_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>, adiar::internal_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
      return __product_construction<prod_policy,
             ^
/Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:111:12: note: in instantiation of function template specialization 'adiar::product_construction<adiar::apply_prod_policy>'
      requested here
    return product_construction<apply_prod_policy>(bdd_1, bdd_2, op);
           ^
In file included from /Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:10:
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:296:28: warning: suggest braces around initialization of subobject [-Wmissing-braces]
          prod_pq_1.push({ r.t1, r.t2, NIL });
                           ^~~~~~~~~~
                           {         }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:604:14: note: in instantiation of function template specialization
      'adiar::__product_construction<adiar::apply_prod_policy, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 1, internal_sorter,
      internal_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>, adiar::internal_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
      return __product_construction<prod_policy,
             ^
/Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:111:12: note: in instantiation of function template specialization 'adiar::product_construction<adiar::apply_prod_policy>'
      requested here
    return product_construction<apply_prod_policy>(bdd_1, bdd_2, op);
           ^
In file included from /Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:10:
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:359:28: warning: suggest braces around initialization of subobject [-Wmissing-braces]
          prod_pq_2.push({ t1, t2, v0.low, v0.high, source });
                           ^~~~~~
                           {     }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:359:28: warning: suggest braces around initialization of subobject [-Wmissing-braces]
          prod_pq_2.push({ t1, t2, v0.low, v0.high, source });
                           ^~~~~~~~~~~~~~~~~~~~~~~
                           {                      }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:102:24: warning: suggest braces around initialization of subobject [-Wmissing-braces]
      prod_pq_1.push({ target.t1, target.t2, source });
                       ^~~~~~~~~~~~~~~~~~~~
                       {                   }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:288:9: note: in instantiation of function template specialization
      'adiar::prod_recurse_out<adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 1, internal_sorter, internal_priority_queue,
      adiar::node_file, 2, std::__1::less<unsigned int>, 1> >' requested here
        prod_recurse_out(prod_pq_1, aw, op, out_uid, r.low);
        ^
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:604:14: note: in instantiation of function template specialization
      'adiar::__product_construction<adiar::apply_prod_policy, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 1, internal_sorter,
      internal_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>, adiar::internal_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
      return __product_construction<prod_policy,
             ^
/Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:111:12: note: in instantiation of function template specialization 'adiar::product_construction<adiar::apply_prod_policy>'
      requested here
    return product_construction<apply_prod_policy>(bdd_1, bdd_2, op);
           ^
In file included from /Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:10:
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:149:24: warning: suggest braces around initialization of subobject [-Wmissing-braces]
      prod_pq_1.push({ r.t1, r.t2, source });
                       ^~~~~~~~~~
                       {         }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:112:19: note: in instantiation of function template specialization
      'adiar::prod_recurse_in__forward::go<adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 1, internal_sorter,
      internal_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1> >' requested here
      out_policy::go(prod_pq_1, aw, ea, prod_pq_1.pull().source);
                  ^
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:397:11: note: in instantiation of function template specialization
      'adiar::prod_recurse_in<adiar::prod_recurse_in__forward, adiar::prod_rec_skipto, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label,
      adiar::tuple_fst_lt, 1, internal_sorter, internal_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>,
      adiar::internal_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
          prod_recurse_in<prod_recurse_in__forward>(prod_pq_1, prod_pq_2, aw, r, t1, t2);
          ^
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:604:14: note: in instantiation of function template specialization
      'adiar::__product_construction<adiar::apply_prod_policy, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 1, internal_sorter,
      internal_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>, adiar::internal_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
      return __product_construction<prod_policy,
             ^
/Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:111:12: note: in instantiation of function template specialization 'adiar::product_construction<adiar::apply_prod_policy>'
      requested here
    return product_construction<apply_prod_policy>(bdd_1, bdd_2, op);
           ^
In file included from /Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:10:
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:296:28: warning: suggest braces around initialization of subobject [-Wmissing-braces]
          prod_pq_1.push({ r.t1, r.t2, NIL });
                           ^~~~~~~~~~
                           {         }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:615:14: note: in instantiation of function template specialization
      'adiar::__product_construction<adiar::apply_prod_policy, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 1, external_sorter,
      external_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>, adiar::external_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
      return __product_construction<prod_policy,
             ^
/Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:111:12: note: in instantiation of function template specialization 'adiar::product_construction<adiar::apply_prod_policy>'
      requested here
    return product_construction<apply_prod_policy>(bdd_1, bdd_2, op);
           ^
In file included from /Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:10:
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:359:28: warning: suggest braces around initialization of subobject [-Wmissing-braces]
          prod_pq_2.push({ t1, t2, v0.low, v0.high, source });
                           ^~~~~~
                           {     }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:359:28: warning: suggest braces around initialization of subobject [-Wmissing-braces]
          prod_pq_2.push({ t1, t2, v0.low, v0.high, source });
                           ^~~~~~~~~~~~~~~~~~~~~~~
                           {                      }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:102:24: warning: suggest braces around initialization of subobject [-Wmissing-braces]
      prod_pq_1.push({ target.t1, target.t2, source });
                       ^~~~~~~~~~~~~~~~~~~~
                       {                   }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:288:9: note: in instantiation of function template specialization
      'adiar::prod_recurse_out<adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 1, external_sorter, external_priority_queue,
      adiar::node_file, 2, std::__1::less<unsigned int>, 1> >' requested here
        prod_recurse_out(prod_pq_1, aw, op, out_uid, r.low);
        ^
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:615:14: note: in instantiation of function template specialization
      'adiar::__product_construction<adiar::apply_prod_policy, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 1, external_sorter,
      external_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>, adiar::external_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
      return __product_construction<prod_policy,
             ^
/Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:111:12: note: in instantiation of function template specialization 'adiar::product_construction<adiar::apply_prod_policy>'
      requested here
    return product_construction<apply_prod_policy>(bdd_1, bdd_2, op);
           ^
In file included from /Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:10:
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:149:24: warning: suggest braces around initialization of subobject [-Wmissing-braces]
      prod_pq_1.push({ r.t1, r.t2, source });
                       ^~~~~~~~~~
                       {         }
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:112:19: note: in instantiation of function template specialization
      'adiar::prod_recurse_in__forward::go<adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 1, external_sorter,
      external_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1> >' requested here
      out_policy::go(prod_pq_1, aw, ea, prod_pq_1.pull().source);
                  ^
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:397:11: note: in instantiation of function template specialization
      'adiar::prod_recurse_in<adiar::prod_recurse_in__forward, adiar::prod_rec_skipto, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label,
      adiar::tuple_fst_lt, 1, external_sorter, external_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>,
      adiar::external_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
          prod_recurse_in<prod_recurse_in__forward>(prod_pq_1, prod_pq_2, aw, r, t1, t2);
          ^
/Users/lasse/Code/adiar/src/adiar/../adiar/internal/product_construction.h:615:14: note: in instantiation of function template specialization
      'adiar::__product_construction<adiar::apply_prod_policy, adiar::levelized_priority_queue<adiar::prod_tuple_1, adiar::tuple_label, adiar::tuple_fst_lt, 1, external_sorter,
      external_priority_queue, adiar::node_file, 2, std::__1::less<unsigned int>, 1>, adiar::external_priority_queue<adiar::prod_tuple_2, adiar::tuple_snd_lt> >' requested here
      return __product_construction<prod_policy,
             ^
/Users/lasse/Code/adiar/src/adiar/bdd/apply.cpp:111:12: note: in instantiation of function template specialization 'adiar::product_construction<adiar::apply_prod_policy>'
      requested here
    return product_construction<apply_prod_policy>(bdd_1, bdd_2, op);
           ^
15 warnings and 1 error generated.
make[5]: *** [src/adiar/CMakeFiles/adiar.dir/bdd/apply.cpp.o] Error 1
make[4]: *** [src/adiar/CMakeFiles/adiar.dir/all] Error 2
make[3]: *** [src/adiar/CMakeFiles/adiar.dir/rule] Error 2
make[2]: *** [adiar] Error 2
make[1]: *** [build/static] Error 2
make: *** [build] Error 2
```

**Comment by SSoelvsten** [comment]
Luckily a power of two is just the same as a left-shift, so `std::pow` can be replaced with `(1 << degree)`.

**Comment by SSoelvsten** [comment]
That looks quite odd. Googling around says one should make sure the visibility is the same across all projects. Both TPIE and Adiar has in their respective *CMakeLists.txt* set the `CXX_VISIBILITY_PRESET` to `hidden`. My best guess would be, that this has something to do with a difference in Boost and how it was compiled?

**Comment by lassemoldrup** [comment]
I am using the Homebrew version of boost (`v. 1.79`), if that helps: https://formulae.brew.sh/formula/boost#default

**Comment by SSoelvsten** [comment]
Ubuntu ships with *1.71* but I cannot remember what version is on my Fedora machine. Can you try to manually change both the settings for *Adiar* and *TPIE* by setting their `CXX_VISIBILITY_PRESET` to `default` instead?

**Comment by lassemoldrup** [comment]
Changing the `CXX_VISIBILITY_PRESET` of just Adiar fixes the warning. I also noticed that one of the warnings generated by CMake is possibly related:
```
CMake Warning (dev) at external/tpie/tpie/CMakeLists.txt:224 (add_library):
  Policy CMP0063 is not set: Honor visibility properties for all target
  types.  Run "cmake --help-policy CMP0063" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  Target "tpie" of type "STATIC_LIBRARY" has the following visibility
  properties set for CXX:

    CXX_VISIBILITY_PRESET

  For compatibility CMake is not honoring them for this target.
This warning is for project developers.  Use -Wno-dev to suppress it.
```

**Comment by SSoelvsten** [comment]
That seems odd - I need to read up on what is going on here. But, thanks for debugging with me. I might ask a few more questions another day.

**Comment by lassemoldrup** [comment]
No problem, feel free.

**Comment by SSoelvsten** [comment]
I created a pull request for TPIE which fixes that very CMake warning (see the pull request above that mentions this issue). Whether that truly fixes this problem I cannot say. What happens if you do the following?
```
cd external/tpie
git remote set-url origin https://github.com/SSoelvsten/tpie.git
git fetch
git checkout cmake/cleanup
cd ../../
make clean
make examples/queens
```
If not, then I guess I have to set `CXX_VISIBILITY_PRESET` for Adiar to `default`?

**Comment by lassemoldrup** [comment]
@SSoelvsten Using your TPIE branch sadly doesn't fix the CMake warning or the linker errors.

**Comment by SSoelvsten** [comment]
So #408 is a fix, @lassemoldrup ?

**Comment by lassemoldrup** [comment]
Yes, that fixes the linker warnings (but not the CMake warning).

**Comment by SSoelvsten** [comment]
What version of *CMake* are you using? The change for TPIE fixed the CMake warning on my version 3.22.2, as far as I could tell.

**Comment by lassemoldrup** [comment]
I am using version 3.24.1.
